lib/improver/tests/test_nbhood_basic.py

```diff
@@ -0,0 +1,497 @@
+     0.98458376, 0.98766701, 0.99280576, 1.0],
+    [1.0, 1.0, 0.99486125, 0.99177801, 0.99075026, 0.99177801,
+     0.99486125, 1.0, 1.0]
+])
+
+
+def set_up_cube(zero_point_indices=((0, 7, 7),), num_time_points=1,
+                num_grid_points=16):
+    """Set up a normal OSGB UK National Grid cube."""
+    data = np.ones((num_time_points, num_grid_points, num_grid_points))
+    for time_index, lat_index, lon_index in zero_point_indices:
+        data[time_index][lat_index][lon_index] = 0
+    cube = Cube(data, standard_name="precipitation_amount",
+                units="kg m^-2 s^-1")
+    coord_system = OSGB()
+    scaled_y_coord = OSGBGRID.coord('projection_y_coordinate')
+    cube.add_dim_coord(
+        DimCoord(
+            scaled_y_coord.points[:num_grid_points],
+            'projection_y_coordinate',
+            units='m', coord_system=coord_system
+        ),
+        1
+    )
+    scaled_x_coord = OSGBGRID.coord('projection_x_coordinate')
+    cube.add_dim_coord(
+        DimCoord(
+            scaled_x_coord.points[:num_grid_points],
+            'projection_x_coordinate',
+            units='m', coord_system=coord_system
+        ),
+        2
+    )
+    tunit = Unit("hours since 1970-01-01 00:00:00", "gregorian")
+    time_points = [402192.5 + _ for _ in range(num_time_points)]
+    cube.add_aux_coord(AuxCoord(time_points,
+                                "time", units=tunit), 0)
+    return cube
+
+
+def set_up_cube_lat_long(zero_point_indices=((0, 7, 7),), num_time_points=1,
+                         num_grid_points=16):
+    """Set up a lat-long coord cube."""
+    data = np.ones((num_time_points, num_grid_points, num_grid_points))
+    for time_index, lat_index, lon_index in zero_point_indices:
+        data[time_index][lat_index][lon_index] = 0
+    cube = Cube(data, standard_name="precipitation_amount",
+                units="kg m^-2 s^-1")
+    cube.add_dim_coord(
+        DimCoord(np.linspace(0.0, float(num_grid_points - 1),
+                             num_grid_points),
+                 'latitude',
+                 units='degrees'),
+        1
+    )
+    cube.add_dim_coord(
+        DimCoord(np.linspace(0.0, float(num_grid_points - 1),
+                             num_grid_points),
+                 'longitude',
+                 units='degrees'),
+        2
+    )
+    tunit = Unit("hours since 1970-01-01 00:00:00", "gregorian")
+    time_points = [402192.5 + _ for _ in range(num_time_points)]
+    cube.add_aux_coord(AuxCoord(time_points,
+                                "time", units=tunit), 0)
+    return cube
+
+
+class Test_operation_radius_to_grid_cells(IrisTest):
+
+    """Test conversion of kernel radius in kilometres to grid cells."""
+
+    RADIUS_IN_KM = 6.1
+
+    def test_basic_radius_to_grid_cells(self):
+        """Test the lat-long radius-to-grid-cell conversion."""
+        cube = set_up_cube()
+        plugin = NBHood(self.RADIUS_IN_KM)
```

Comment:
Just checking, are we still continuing with the "plugin" terminology?
We are no longer using it in the main routines, but are using it in the unit-tests.